### PR TITLE
added typechecks

### PR DESCRIPTION
**Pull request by jbloom22**
- genotype.py
- variant.py
- interval.py
- annotations.py

### CODE REVIEW

**Comment by jbloom22** [comment]
Fixed #1832

File: python/hail/representation/genotype.py [hunk 1]
```diff
@@ -295,6 +297,7 @@ def one_hot_genotype(self, num_genotypes):
         return jiterable_to_list(from_option(self._jrep.oneHotGenotype(num_genotypes)))
 
     @handle_py4j
+    @typecheck_method(theta=nullable(numeric))
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: I think the docs are wrong here. This isn't nullable. Can you make this just numeric and fix the docstring?

File: python/hail/representation/interval.py [hunk 2]
```diff
@@ -122,6 +124,7 @@ def contains(self, locus):
         return self._jrep.contains(locus._jrep)
 
     @handle_py4j
+    @typecheck_method(interval=Interval)
```
Comment (tpoterba) [inline python/hail/representation/interval.py]: does this work? I thought this is a python name error. If it crashes the tests, change this to `anytype` for now.

Comment (jbloom22) [inline python/hail/representation/interval.py]: changed. back to you

File: python/hail/representation/interval.py [hunk 3]
```diff
@@ -49,6 +49,7 @@ def _from_java(cls, jrep):
 
     @staticmethod
     @handle_py4j
+    @typecheck_method(string=strlike)
```
Comment (tpoterba) [inline python/hail/representation/interval.py]: should be @typecheck, not @typecheck_method

File: python/hail/representation/variant.py [hunk 4]
```diff
@@ -445,6 +446,7 @@ def _from_java(cls, jrep):
 
     @staticmethod
     @handle_py4j
+    @typecheck_method(string=strlike)
```
Comment (tpoterba) [inline python/hail/representation/variant.py]: static = not method
@typecheck

**Comment by jbloom22** [comment]
back to you

**Pull request merged**